workitem/workitem_repository.go

```diff
@@ -104,6 +104,10 @@ func (r *GormWorkItemRepository) Save(ctx context.Context, wi app.WorkItem) (*ap
 	if res.Version != wi.Version {
 		return nil, errors.NewVersionConflictError("version conflict")
 	}
+	// Changing the type of a work item is prohibited.
+	if res.Type != wi.Type {
+		return nil, errors.NewBadParameterError("type", wi.Type).Expected(res.Type)
```

Comment:
@aslakknutsen okay, after our discussion I've modified the code to silently ignore WIT type changes instead of causing a BadRequest error. See d83394a